### PR TITLE
Fixes #15 use full reference to variable names

### LINKED ISSUE #15: Don't use 'inherit' in sub-classes
This isn't considered good practice.
See https://docs.puppet.com/puppet/4.10/lang_classes.html#inheritance

### PR DESCRIPTION
**Pull request by traylenator**
As per style guide and puppetlabs-ntp avoid use
of inherits.

### CODE REVIEW

File: templates/nscd.conf.erb [hunk 1]
```diff
@@ -34,48 +34,48 @@
 
 
 #	logfile			/var/log/nscd.log
-	threads			<%= @threads %>
+	threads			<%= scope['nscd::threads'] %>
```
Comment (juniorsysadmin) [inline templates/nscd.conf.erb]: Short variable names are better in templates, hence the inherits for config.pp

**Comment by traylenator** [comment]
@juniorsysadmin I kind of agree but it's not the reality of:

https://github.com/puppetlabs/puppetlabs-ntp/blob/master/templates/keys.epp#L3
or
https://puppet.com/docs/puppet/5.3/style_guide.html#class-inheritance

**Comment by traylenator** [comment]
Converting to .epp and passing in variables is probably cleaner.

**Comment by traylenator** [comment]
@juniorsysadmin you have a moment?

File: manifests/config.pp [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
-#
-class nscd::config inherits nscd {
+# @api private
+# This class handles the configuration file. Avoid modifying private classes.
+class nscd::config {
   file{'/etc/nscd.conf':
     ensure  => present,
-    content => template('nscd/nscd.conf.erb'),
+    content => epp('nscd/nscd.conf.epp', {
+      'threads'               => $nscd::threads,
```
Comment (juniorsysadmin) [inline manifests/config.pp]: Why not use `$nscd::threads` for example inside the epp instead?

Comment (traylenator) [reply manifests/config.pp]: To address your earlier comment:

    Short variable names are better in templates, hence the inherits for config.pp

Comment (juniorsysadmin) [reply manifests/config.pp]: Hmm I don't think I was very clear here.
Without `inherits` in the class, in ERB templates you often have things like `scope['nscd::threads']` or `scope.lookupvar('nscd::threads')` - this is long and ugly.
But without `inherits` in the class, in EPP templates you can use the full reference within the template like `threads <%= $nscd::threads %` which is not really that long, and reasonably neat. It seems Puppet has done it that way in their modules like https://github.com/puppetlabs/puppetlabs-ntp/blob/master/templates/ntp.conf.epp . I couldn't find anything in the style guide about referencing variables within EPP templates however.

Comment (traylenator) [reply manifests/config.pp]: Makes sense, I went past what you suggested. Agreed its the scope thing that is ugly not the `$::nscd::threads`
I'll change it. Less annoying code to add when adding a new variable.

**Comment by traylenator** [comment]
@juniorsysadmin after doing some more .epp I think I understand the best practices now . Will update shortly.

**Comment by bastelfreak** [comment]
ping @traylenator :)

**Comment by traylenator** [comment]
Was thinking everything was blocked - not this one. Okay.

**Comment by traylenator** [comment]
Close in favour of #51

**Pull request closed without merging**